File: week8/javascript/bm4656/_9536.js
//00:30:36
function solution(data) {
  let [T, ...input] = data.toString().trim().split('\n');
  let answer = '';
  let arr = [];
  for (let i = 0; i < input.length; i++) {
    if (input[i] === 'what does the fox say?') {
      let [all, ...animals] = arr;
      // console.log(all, animals);
      let cry = [];
      all = all.split(' ');
      for (let i = 0; i < animals.length; i++) {
        cry.push(animals[i].split(' ')[2]);
      }
      all = all.filter(x => !cry.includes(x));
      // console.log(cry);
      answer = all.join(' ');
      arr = [];
      continue;
    } else arr.push(input[i]);
  }
  return answer;
}

const case1 = `1
toot woof wa ow ow ow pa blub blub pa toot pa blub pa pa ow pow toot
dog goes woof
fish goes blub
elephant goes toot
seal goes ow
what does the fox say?`;

console.log(solution(case1));

File: week8/java/src/Gawq/_9536.java
package Gawq;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.StringReader;
import java.util.ArrayList;
import java.util.StringTokenizer;

public class _9536 {
    public static void main(String[] args) throws IOException {
        BufferedReader input = new BufferedReader(new StringReader(src));

        int T = Integer.parseInt(input.readLine());
        ArrayList<String> list = new ArrayList<>();
        ArrayList<String> cry = new ArrayList<>();
        StringTokenizer st;

        for (int i = 0; i < T; i++) {
            st = new StringTokenizer(input.readLine());
            while(st.hasMoreTokens()) {
                list.add(st.nextToken());
            }

            while (true) {
                String s = input.readLine();
                st = new StringTokenizer(s);
                if (s.equals("what does the fox say?")) break;
                for (int j = 0; j < 3; j++) {
                    if (j == 2) {
                        cry.add(st.nextToken());
                    }
                    else st.nextToken();
                }
            }

            for (int k = 0; k < list.size(); k++) {
                for (int l = 0; l < cry.size(); l++) {
                    if (list.get(0).equals(cry.get(l))) {
                        list.remove(0);
                    }
                    if (list.get(k).equals(cry.get(l))) {
                        list.remove(k);
                        k = 0;
                        break;
                    }
                }
            }

            for (String s : list) {
                System.out.print(s + " ");
            }
        }

    }

    private static String src = "1\n" +
            "meow toot woof meow wa ow ow meow ow pa blub blub pa toot meow meow pa blub pa pa ow pow toot\n" +
            "dog goes woof\n" +
            "fish goes blub\n" +
            "elephant goes toot\n" +
            "seal goes ow\n" +
            "cat goes meow\n" +
            "what does the fox say?";
}
